docs/contributor-guide/component/identifier.md

```diff
@@ -1,14 +1,20 @@
 # Registering Identifier Patterns
-When [writing unpackers](./unpacker.md), OFRAK Contributors can leverage the `MagicMimeIdentifier` and `MagicDescriptionIdentifier` by registering mappings between resource tags and mime or description patterns. Doing so will ensure that `Resource.unpack` automatically calls their custom unpacker.
+OFRAK Contributors should implement an [Identifier][ofrak.component.identifier.Identifier] to automatically identify and label resources with a [ResourceTag][ofrak.model.tag_model.ResourceTag].
 
-For example, consider the following magic description identification registration in the file containing a `UImageUnpacker`:
+First, consider leveraging the [MagicIdentifier][ofrak.core.magic.MagicIdentifier] by registering a pattern match with either the [MagicMimePattern][ofrak.core.magic.MagicMimePattern], [MagicDescriptionPattern][ofrak.core.magic.MagicDescriptionPattern], or [RawMagicPattern][ofrak.core.magic.RawMagicPattern].
+
+For example, consider the following magic description identification registration for [UImage][ofrak.core.uimage.UImage]:
 
 ```python
-MagicDescriptionIdentifier.register(UImage, lambda s: s.startswith("u-boot legacy uImage"))
+MagicDescriptionPattern.register(
+    UImage, lambda s: s.startswith("u-boot legacy uImage")
+)
 ```
 
-This line ensures that the `MagicDescriptionIdentifier` adds a `UImage` tag to resources matching that description pattern. As a result, any unpackers targeting a `UImage` will automatically run when `Resource.unpack` is run.
+This pattern (along with all other identifier patterns) will get run when the [MagicIdentifier][ofrak.core.magic.MagicIdentifier] runs, resulting in a `UImage` tag to resources matching the description.
 
+Registering magic patterns is a good first choice, since it is the most performance-efficient.
+When required, it is also possible to write custom identifiers. For an example of this, see [ApkIdentifier][ofrak.core.apk.ApkIdentifier].
```

Comment:
We should add more docs here with examples of the other two magic patterns. For example, there are some aspects of the raw bytes pattern matcher that are pretty unintuitive (see comment below).

This is also a good place to add information about when to use patterns and when not to (perhaps pointing to the APK as an example of when the magic pattern thing isn't quite sufficient).

Reply:
Updated documentation.